### ISSUE TITLE
Verify taxa: return correct taxa df if input already contains verificationKey

### ISSUE THREAD

**Issue by peterdesmet** [trias-project/trias#30]
If the input `taxa` already contains a `verificationKey`, then the function works but returns a `taxa` df with more (duplicated) columns:

```R
# ... with 6,097 more rows, and 21 more variables:
#   bb_kingdom <chr>, bb_rank <chr>, bb_taxonomicStatus <chr>,
#   bb_acceptedKey <dbl>, bb_acceptedName <chr>,
#   verificationKey.x.x <chr>, verificationKey.x <lgl>,
#   taxonID <chr>, nameType <chr>, issues <chr>,
#   validDistribution <lgl>, bb_species <chr>, bb_genus <chr>,
#   bb_family <chr>, bb_order <chr>, bb_class <chr>,
#   bb_phylum <chr>, bb_speciesKey <dbl>,
#   verificationKey.y <lgl>, verificationKey.y.y <chr>,
#   verificationKey <chr>
```

Ideally, the df is return exactly like the input `taxa`, with an updated `verifiedKey` column or that column appended to the end.

**Comment by peterdesmet** [comment]
Steps I currently take to circumvent this:

https://github.com/trias-project/unified-checklist/blob/b91446ba491f2aed77e0d5ef303108f6b554c248/src/verify_taxa.Rmd#L17

https://github.com/trias-project/unified-checklist/blob/b91446ba491f2aed77e0d5ef303108f6b554c248/src/verify_taxa.Rmd#L50-L68

**Comment by damianooldoni** [comment]
Strange error. I will investigate it.

**Comment by damianooldoni** [comment]
Actually not a strange error! 
I developed the function assuming that input `taxa` didn't have any `verificationKey` column at all. So strange things happened.
The point is: what's the need of `verificationKey` in **input** `taxa`? 
The `verificationKey` is set equal to `bb_taxonKey` (taxonomic status `ACCEPTED` or `DOUBTFUL`) or it is stored in `verification` df (second input of function). In both cases we would overwrite the values contained in `taxa`.
So, we can remove this column. We can do it:
1. in pipeline just before calling `verify_taxa()`
2. in function `verify_taxa` as very first step.

In both cases something like this:
```r
taxa <- taxa %>% select(-verificationKey)
``

@peterdesmet : did I miss something? You probably opt for second solution, do you?

**Comment by peterdesmet** [comment]
A verificationKey in input might be the result of rerunning the output. Since it can lead to unexpected output, I would handle it in the function.

**Comment by damianooldoni** [comment]
Ok. I will add `taxa %>% select(-verificationKey)` in the function.
By doing so, the only verification keys which can be changed without any informative message will be due to new relations `taxonKey` -  `bb_key` of accepted or doubtful taxa.  Do you want take trace of these changes? I think so.

**Comment by peterdesmet** [comment]
Not sure I follow which changes specifically you want to report on.

**Comment by damianooldoni** [comment]
Yes, you don't follow it because I cannot provide such information :upside_down_face: 
`my_verification` contains only synonyms and unmatched taxa, it doesn't track changes in `taxonKey` - `bb_key` coupling of accepted or doubtful taxa. :+1:

**Comment by damianooldoni** [comment]
This issue can be closed based on closed PR #32.
